whitepapers/tickets/SHARE-822.md

```diff
@@ -0,0 +1,112 @@
+   - Continue to use the Bepress taxonomy as SHARE's central taxonomy
+
+
+## Changes to SHARE:
+  - New Model: "Taxonomy"
+  - Changes to Model: "Subject"
+    - Unique indexes on Subjects will move from `[name]` to `[name, taxonomy_id]` if the OSF requirements are ammenable to this
+      Otherwise it will become `[name, taxonomy_id, parent_id]`
+  - If we can upgrade to elasticsearch 5.4+ in time SHARE will index subjects using the new Path Hierarchy Tokenizer.
+    Otherwise subjects will be manually indexed to act like the Path Hierarchy Tokenizer
+  - Subjects will be indexed with the first element as the name of their Taxonomy to namespaces them appropriately
+    IE: Bepress/A/B/C or Engrxiv/A/B/C
+  - The Elasticsearch task will be updated to look for any related models that have been recently updated as well
+  - A page will be added to the admin interface to support manually fixing Taxonomies if required
+
+### Taxonomy
+
+| Column        |   Type   | Indexed | Nullable | FK  | Default | Description                                    |
+| :------------ | :------: | :-----: | :------: | :-: | :-----: | :--------------------------------------------- |
+| name          |   text   | unique  |          |     |         | The human readable name for this taxonomy      |
+| deleted       |   bool   |         |          |     |         | Whether or not this taxonomy is deleted/usable |
+| date_created  | datetime |         |          |     |         |                                                |
+| date_modified | datetime |  Desc   |          |     |         |                                                |
+
+| id  | name          | deleted |
+| --- | ------------- | ------- |
+| 1   | Bepress       | False   |
+| 1   | My custom tax | False   |
+
+## Subject
+
+| Column        |   Type   | Indexed | Nullable | FK  | Default | Description                                                                                        |
+| :------------ | :------: | :-----: | :------: | :-: | :-----: | :------------------------------------------------------------------------------------------------- |
+| name          |   text   |         |          |     |         | The subject name                                                                                   |
+| deleted       |   bool   |         |          |     |         |                                                                                                    |
+| handle        |   text   | unique  |          |     |         | A unique refence to this subject that Sources may use to specify which subject needs to be altered |
+| taxonomy_id   |   int    |    ✓    |          |  ✓  |         | The taxonomy this subject belongs to                                                               |
+| alias_id      |   int    |    ✓    |          |  ✓  |         | The Bepress equivalent of this subject. Set to id if this subject belongs to Bepress.              |
```

Comment:
I think not using `bepress_id` is a good idea although maybe not `alias_id`. Not that I have a better idea... maybe `default_id` or `master_id`